### DISCUSSION TITLE
what is the string escape syntax for duckdb

### DISCUSSION THREAD

**Discussion by junranhe** [duckdb/duckdb#8992] [Q&A]
Hello everyone, I try to use duckdb v0.81 for instead mysql (I know the syntax is very different)

Can Duckdb use backslashes ‘\\' for string escape?
If not, what is the string escape syntax for duck db?
how can I express the single-queto “ ‘ “ in the duckdb string express?
I can not find any documents in data type “text" about this 

SQL: SELECT '[{ \\"name \\":  \\"true\\"}]' AS f in MySQL
The string data returned using jdbc's resultset is:
 [{"name": "true”}]  (data not contains “\\")

But the same SQL in duck db:
The string data returned using jdbc's resultset is:
 [{ \”name\ “:  \\"true\\"}]

**Comment by Tishj** [comment]
This creates a character literal:
```
'hello'
```

To create a character literal that contains `'` they need to be doubled, for example:
```
'hello ''world''!'
```

**Comment by l1t1** [comment]
same  syntax  as Oracle, and Oracle also has q[   ]

**Reply by junranhe** [reply]
I see， but how can I make string with “ \r\n”?  My sql string is a “javascript” file data, like ' var a = 10 \\r\\n ...  ' , if use oracle syntax, then '\\r\\n' will be ''' || chr(10)||chr(13) ||''' ， this expression is  hard to understand.

**Reply by Tishj** [reply]
In DuckDB you would need to use the same functions to create the carriage return + newline combo, there is no support for escaping characters, everything is interpreted literally
